src/fsharp/symbols/Exprs.fs

```diff
@@ -530,11 +586,67 @@ module FSharpExprConvert =
                 let argR = ConvExpr cenv env arg
                 E.ILFieldSet(None, typR, fspec.Name, argR) 
 
+            | TOp.ILAsm([ ], _), _, [arg] -> 
+                ConvExprPrim cenv env arg
+
+            | TOp.ILAsm([ I_box _ ], _), [ty], [arg] -> 
+                let op = mkCallBox cenv.g m ty arg
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ I_unbox_any _ ], _), [ty], [arg] -> 
+                let op = mkCallUnbox cenv.g m ty arg
+                ConvExprPrim cenv env op
 
-            | TOp.ILAsm([ AI_ceq ], _), _, [arg1;arg2]  -> 
+            | TOp.ILAsm([ I_isinst _ ], _), [ty], [arg] -> 
+                let op = mkCallTypeTest cenv.g m ty arg
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ I_ldtoken _ ], _), [ty], _ -> 
+                let op = mkCallTypeOf cenv.g m ty
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ EI_ilzero _ ], _), [ty], _ -> 
+                E.DefaultValue (ConvType cenv ty)
+
+            | TOp.ILAsm([ AI_ldnull; AI_cgt_un ], _), _, [arg] -> 
+                let elemTy = tyOfExpr cenv.g arg
+                // let op = mkCallIsNotNull cenv.g m elemTy arg
+                let nullVal = mkNull m elemTy
+                let op = mkCallNotEqualsOperator cenv.g m elemTy arg nullVal
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ I_ldlen; AI_conv DT_I4 ], _), _, [arr] -> 
+                let arrayTy = tyOfExpr cenv.g arr
+                let elemTy = destArrayTy cenv.g arrayTy
+                let op = mkCallArrayLength cenv.g m elemTy arr
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ I_newarr (ILArrayShape [(Some 0, None)], _)], _), [elemTy], xa ->
+                E.NewArray(ConvType cenv elemTy, ConvExprs cenv env xa)
+
+            | TOp.ILAsm([ I_ldelem_any (ILArrayShape [(Some 0, None)], _)], _), [elemTy], [arr; idx1]  -> 
+                let op = mkCallArrayGet cenv.g m elemTy arr idx1
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ I_stelem_any (ILArrayShape [(Some 0, None)], _)], _), [elemTy], [arr; idx1; v]  -> 
+                let op = mkCallArraySet cenv.g m elemTy arr idx1 v
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ ILUnaryOp unaryOp ], _), _, [arg] -> 
+                let ty = tyOfExpr cenv.g arg
+                let op = unaryOp cenv.g m ty arg
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ ILBinaryOp binaryOp ], _), _, [arg1;arg2] -> 
                 let ty = tyOfExpr cenv.g arg1
-                let eq = mkCallEqualsOperator cenv.g m ty arg1 arg2
-                ConvExprPrim cenv env eq
+                let op = binaryOp cenv.g m ty arg1 arg2
+                ConvExprPrim cenv env op
+
+            | TOp.ILAsm([ ILConvertOp _ ; ILConvertOp convertOp ], _), _, [arg]
```

Comment:
These double-conversion ops are a bit slippery - we need systematic testing for all the different cases where these are produced in prim-types.fs?

Reply:
@dsyme Added dual conversions with tests.